servers/src/main/java/tachyon/worker/block/BlockMasterSync.java

```diff
@@ -220,4 +220,33 @@ private void resetMasterClient() {
         new MasterClient(NetworkAddressUtils.getMasterAddress(mTachyonConf),
             mMasterClientExecutorService, mTachyonConf);
   }
+
+  /**
+   * Thread to remove block from master
+   */
+  private class BlockRemover implements Runnable {
+    private BlockDataManager mBlockDataManager;
+    private long mUserId;
+    private long mBlockId;
+
+    public BlockRemover(BlockDataManager blockDataManager, long userId, long blockId) {
+      mBlockDataManager = blockDataManager;
+      mUserId = userId;
+      mBlockId = blockId;
+    }
+
+    @Override
+    public void run() {
+      try {
+        mBlockDataManager.removeBlock(mUserId, mBlockId);
+      } catch (IOException ioe) {
```

Comment:
If an exception do thrown here, logically it will happen in the order: `InvalidStateException`, `IOException`, `NotFoundException`, the original `catch` structure is okay, better adjust it as described above. 

By the way, the `IOException` may be thrown delete to **block deleted fails** (see `TieredBlockStore#removeBlockInternal`), better improve the `LOG.warn` log message corresponding.

Thanks!